scripts/run_e2e_tests.py

```diff
@@ -320,20 +319,32 @@ def main(args=None):
                 break
 
             # Check whether we should rerun based on this suite's policy.
-            test_is_flaky = flake_checker.is_test_output_flaky(
+            test_is_flaky, rerun = flake_checker.is_test_output_flaky(
                 output, parsed_args.suite)
-            if policy == RERUN_POLICY_NEVER:
-                print(
-                    'Not rerunning because the policy is to never '
-                    'rerun the {} suite'.format(parsed_args.suite))
-                break
-            if policy == RERUN_POLICY_KNOWN_FLAKES and not test_is_flaky:
-                print((
-                    'Not rerunning because the policy is to only '
-                    'rerun the %s suite on known flakes, and this '
-                    'failure did not match any known flakes')
-                    % parsed_args.suite)
-                break
+            if rerun != flake_checker.RERUN_UNKNOWN:
+                if rerun == flake_checker.RERUN_YES:
+                    print(
```

Comment:
If I'm understanding correctly: This `rerun` overrides RERUN_POLICY defined above, right? If yes, then is any specific reason for using a separate RERUN_POLICY here? (If the rerun check is handled in the flake-checker-server then can we have single-source-of-truth [not sure!])

Reply:
Eventually we will have all of this handled by the flake-checker server, but since oppia/oppia and the flake-checker won't be updated at the same time, we need a transitional state that's backwards-compatible. That's what this PR provides. Once the logger server is updated, we can move the rerun policies over and remove them from here

Reply:
is it possible to do changes to this file once the flake-checker repo and the server are ready? (I think the rerun related PR on flake-checker repo got merged) OR maybe have different API URLs with version tags? e.g, `/v1/check-flake` (This will help us to follow a similar pattern in the future for backward compatibility? (not sure!)

WDYT?

Reply:
The rerun-related PR on the flake-checker repo has merged, but it doesn't include the policies. We're trying to fast-track this PR so that we can rerun on the renderer timeout flake since we haven't been able to fix that flake, but it's annoying for developers when that flake occurs on suites with a "never" rerun policy. Putting the rerun policies in the flake-checker repo is a little more complicated, so I don't want that work to block us from fixing this annoyance for developers quickly.

Having different API URLs is definitely possible, but it would take substantial extra work that I'd rather put towards fixing the E2E flakes right now. Since we only have one client for this server and we control both the client and the server, I don't think we need to get too fancy with API versioning

Reply:
> Since we only have one client for this server and we control both the client and the server, I don't think we need to get too fancy with API versioning

My concern is about streamlining the workflow between these two projects i.e, finding a way to make the flake-check development independent and should not affect the oppia build.

I'm fine with getting this PR in but we should definitely find a way to make flake-checker development simple and avoid complicated code in oppia repo. 

Let's leave a code comment on top (Around line 321) explaining that the rerun policy check will be moved completely to the other repo in the future (in `scripts/run_e2e_tests.py` file). wdyt?